pyvista/core/filters.py

```diff
@@ -1599,14 +1599,58 @@ def decimate_boundary(dataset, target_reduction=0.5):
         return dataset.extract_geometry().triangulate().decimate(target_reduction)
 
 
+    def sample_over_line(dataset, pointa, pointb, resolution=None, scalars=None):
+        """Sample a dataset along a high resolution line.
+
+        Parameters
+        ----------
+        pointa : np.ndarray or list
+            Location in [x, y, z].
+
+        pointb : np.ndarray or list
+            Location in [x, y, z].
+
+        resolution : int
+            number of pieces to divide line into. Defaults to number of cells
+            in the input mesh. Must be a positive integer.
+
+        scalars : str
+            The string name of the variable in the input dataset to probe. The
+            active scalar is used by default.
+
+        Return
+        ------
+        distance : np.ndarray
+            The distance along the line at the sampled points.
+        
+        values : np.ndarray
+            The values of scalars at the sampled points
+        """
+        if resolution is None:
+            resolution = int(dataset.n_cells)
+        if not isinstance(resolution, int) or resolution < 0:
+            raise RuntimeError('`resolution` must be a positive integer, not {}'.format(type(resolution)))
+        # Make a line and probe the dataset
+        line = pyvista.Line(pointa, pointb, resolution=resolution)
+        sampled = line.sample(dataset)
+
+        # Get variable of interest
+        if scalars is None:
+            _, scalars = dataset.active_scalars_info
+        values = sampled.get_array(scalars)
+        distance = sampled['Distance']
+
+        return distance, values
```

Comment:
I think this filter should return the `PolyData` line that is sampled. This would allow you to then sample all of the point data in the input `dataset` along the line. Thus, just make the retur statement here

```py
        return line.sample(dataset)
```

Reply:
and it would make this filter more consistent with the `sample` filter as this method is a really just a helper for the filter